### PR TITLE
Implemented click on element step

### PR DESCRIPTION
**Pull request by amcpanaligan**
# Notes
1. Implemented a step that takes a `domQuerySelector` and attempts to click that element

### CODE REVIEW

File: src/steps/click-on-element.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { BaseStep, Field, StepInterface } from '../core/base-step';
+import { Step, RunStepResponse, FieldDefinition, StepDefinition } from '../proto/cog_pb';
+
+export class ClickOnElement extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Click an element on a page';
+  // tslint:disable-next-line:max-line-length
+  protected stepExpression: string = 'click on page element (?<domQuerySelector>.+)';
```
Comment (iamEAP) [inline src/steps/click-on-element.ts]: @amcpanaligan Reading this again, I think `click on` sounds a little old-fashioned.  Perhaps we just use `click the page element`

**Comment by amcpanaligan** [comment]
> @amcpanaligan may I recommend something like...
> 
> ```yaml
> steps:
> - step: Given I navigate to https://crank.run
> - step: When I click on page element .homepage-hero a[href="/intro"]
> - step: Then the title meta tag on the current page should be Crank - Introduction
> ```
> 
> Kind of a meaningless test, but not completely unwarranted, given [how Gatsby does internal navigation](https://www.gatsbyjs.org/docs/gatsby-link/) / routing / pre-fetching ([e.g.](https://github.com/run-crank/run-crank.github.io/blob/7b44bf1/src/components/homepage-hero/index.js#L10))

This is smart. It seems that my mind is still scoped to that specific page (Just as we do form test in AT) and was not able to think of assertions outside that current page.

**Comment by amcpanaligan** [comment]
It is weird that if you put a wait in the step puppeteer does not detect the url change

This fails:
image

**Comment by amcpanaligan** [comment]
> It is weird that if you put a wait in the step puppeteer does not detect the url change
> 
> This fails:
> image

It seems the url is not changing. But page meta such as `title` changes. So I got my test passing by asserting it instead of the `url`. I am not sure if it's a separate issue that it doesn't change in puppeteer's context

**Comment by iamEAP** [comment]
@amcpanaligan Yes, that might be worth filing a bug for (unrelated to this, naturally).  Right now, we only update the response object (on which page URL, status, and text rely) after the page navigation step.

**Pull request merged**